pages/Configuring/Uncommon-tips-&-tricks.md

```diff
@@ -114,6 +114,66 @@ bind = $mod, S, movetoworkspace, special:magic
 bind = $mod, S, togglespecialworkspace, magic
 ```
 
+## Show desktop
+
+This approach uses same principle as the `Minimize windows using special workspaces` section.
+It moves all windows from current workspace to a `special` workspace named `desktop`.
+Showing desktop state is remembered per workspace.
+
+Create a script:
+```sh
+#!/usr/bin/env sh
+
+TMP_FILE="/tmp/hyprland-show-desktop"
+
+NUMBER_HIDDEN_WINDOWS=$(hyprctl workspaces -j | jq '.[] | select (.name=="special:desktop") | .windows')
+
+CW=$(hyprctl monitors -j | jq '.[] | .activeWorkspace | .name' | sed 's/"//g')
+
+echo $CW
+echo $NUMBER_HIDDEN_WINDOWS
+
+if [ -s "$TMP_FILE-$CW" ]; then
+  readarray -d $'\n' -t AA <<< $(< "$TMP_FILE-$CW")
+
+  for address in "${AA[@]}"
+  do
+    echo "HERE"
+    cmd="hyprctl dispatch movetoworkspacesilent name:$CW,address:$address"
+    eval $cmd
+  done
+
+  rm "$TMP_FILE-$CW"
+else
+  COMMAND="hyprctl clients -j | jq '.[] | select (.workspace .name == \"$CW\")'"
+  echo $COMMAND
+  NHW="$( eval $COMMAND | jq '.address' )"
+ 
+  readarray -d $'\n' -t AA <<< $NHW
+
+  TMP_FILES=""
```

Comment:
Redundant? or did you mean `TMP_FILE`?

Reply:
I am pretty sure I meant to write TMP_ADDRESS, couse my bash scripting is not that good and idk the scope of the if statement. but yeah it is redundant.